### PR TITLE
Feature/analysis graphql

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- [x] Migrate Analysis from REST to Graphql - Partial
- [x] #1289
- [x] https://github.com/the-deep/server/issues/1275
- [x] https://github.com/the-deep/server/issues/1276
- https://github.com/the-deep/server/issues/1280
  - [x] https://github.com/the-deep/server/issues/1281
  - [x] https://github.com/the-deep/server/issues/1282
  - [x] https://github.com/the-deep/server/issues/1283
  - [ ] https://github.com/the-deep/server/issues/1284

Depends on
- https://github.com/the-deep/deeper/pull/983

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [ ] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/analysis/mutation.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    def get_serializer_context(cls, instance, context):
+        return {
+            **context,
+            'analysis_end_date': instance.analysis.end_date,
+        }
+
+
+class CreateAnalysisPillarDiscardedEntry(DiscardedEntriesMutationMixin, PsGrapheneMutation):
+    class Arguments:
+        id = graphene.ID(required=True)
```
Comment (k9845) [inline apps/analysis/mutation.py]: Why `id` in creation?

File: apps/analysis/mutation.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        data = DiscardedEntryCreateInputType(required=True)
+    model = DiscardedEntry
+    serializer_class = DiscardedEntryGqlSerializer
+    result = graphene.Field(AnalysisPillarDiscardedEntryType)
+
+
+class UpdateAnalysisPillarDiscardedEntry(DiscardedEntriesMutationMixin, PsGrapheneMutation):
+    class Arguments:
+        data = DiscardedEntryUpdateInputType(required=True)
```
Comment (k9845) [inline apps/analysis/mutation.py:121]: Missing `id` here

File: apps/analysis/schema.py [hunk 3]
```diff
@@ -1,71 +1,123 @@
...
+    # XXX: N+1 and No pagination
+    entries = graphene.List(graphene.NonNull(AnalyticalStatementEntryType))
+
+    @staticmethod
+    def get_custom_queryset(queryset, info, **_):
+        return get_analytical_statement_qs(info)
+
+    @staticmethod
+    def resolve_entries_count(root, info, **_):
+        return info.context.dl.analysis.analysis_statement_analyzed_entries.load(root.id)
```
Comment (k9845) [inline apps/analysis/schema.py]:
```suggestion
        return info.context.dl.analysis.analytical_statement_analyzed_entries.load(root.id)
```

File: apps/analysis/serializers.py [hunk 4]
```diff
@@ -220,3 +220,188 @@ class Meta:
...
+            'order',
+            'cloned_from',
+            # Custom
+            'entries',
+            'client_id',
+        )
+
+    # NOTE: This is a custom function (apps/user_resource/serializers.py::UserResourceSerializer)
+    # This makes sure only scoped (individual AnalyticalStatement) instances (entries) are updated.
+    # For Secondary tagging
```
Comment (k9845) [inline apps/analysis/serializers.py]:
```suggestion
```

File: apps/entry/dataloaders.py [hunk 5]
```diff
@@ -17,6 +17,16 @@
...
 
 
+class EntryLoader(DataLoaderWithContext):
+    def batch_load_fn(self, keys):
+        entry_qs = Entry.objects.filter(id__in=keys)
+        _map = {
+            entry.id: entry
+            for entry in entry_qs
+        }
+        return Promise.resolve([_map[key] for key in keys])
```
Comment (bewakes) [inline apps/entry/dataloaders.py:28]: Maybe do `for key in _map.keys()`? or `_map.get(key, <some default>`. There's a possibility that `entry_qs` won't have all items in keys.

File: apps/deepl_integration/handlers.py [hunk 6]
```diff
@@ -0,0 +1,810 @@
...
+
+        common_attrs = dict(
+            model_version=model_version,
+            draft_entry_id=draft_entry.id,
+        )
+        new_predictions = []
+        for category_tag, tags in tags.items():
+            for tag, prediction_data in tags.items():
+                prediction_value = prediction_data.get('prediction')
+                threshold_value = prediction_data.get('prediction')
```
Comment (bewakes) [inline apps/deepl_integration/handlers.py]: Looks like a typo, it should probably be `.get('threshold')` right?

File: apps/deepl_integration/handlers.py [hunk 7]
```diff
@@ -0,0 +1,810 @@
...
+    def send_trigger_request_to_extractor(cls, connector_leads_qs: models.QuerySet[ConnectorLead]):
+        paginator = Paginator(
+            connector_leads_qs.filter(
+                extraction_status=ConnectorLead.ExtractionStatus.PENDING
+            ).only('id', 'url').order_by('id').distinct(),
+            100,
+        )
+        processed = 0
+        while True:
+            page = paginator.page(1)
```
Comment (bewakes) [inline apps/deepl_integration/handlers.py:562]: This line, as it seems to me, will probably cause an infinite loop. The page number`(1)` does not seem to get updated. Let me know if I am wrong.

File: deep/token.py [hunk 8]
```diff
@@ -18,6 +19,9 @@ class DeepTokenGenerator(PasswordResetTokenGenerator):
     class Meta:
         abstract = True
 
+    def make_token(self, instance: models.Model):
```
Comment (bewakes) [inline deep/token.py]: Isn't this definition redundant?

**Review by bewakes — CHANGES_REQUESTED** [review]

File: apps/entry/dataloaders.py (hunk 5 above)
Comment (thenav56) [reply apps/entry/dataloaders.py:28]: keys are generated from current existing entries, so _map will always be available for each key.

File: apps/deepl_integration/handlers.py (hunk 6 above)
Comment (thenav56) [reply apps/deepl_integration/handlers.py]: Nice catch 👍🏽

File: apps/deepl_integration/handlers.py (hunk 7 above)
Comment (thenav56) [reply apps/deepl_integration/handlers.py:562]: Added a comment on why.

File: deep/token.py (hunk 8 above)
Comment (thenav56) [reply deep/token.py]: This is just for type definition for now.

**Review by bewakes — APPROVED** [review]

**Pull request merged**